doc/api/screen.rst

```diff
@@ -0,0 +1,120 @@
+communicate with a user. See the :ref:`screen handling <screen_handling_label>` section to find
+out more.
+
+The methods :meth:`UIScreen.redraw`, :meth:`UIScreen.close`, :meth:`UIScreen.emit` and
+:meth:`UIScreen.create_and_emit` are asynchronous. These methods will create a signal which
+is passed to the event loop for later processing.
+
+**Beware**, methods :meth:`UIScreen.redraw` and :meth:`UIScreen.close` can lead to unexpected
+behavior when multiple instances of these signals are emitted (methods are called multiple times).
+This can even crash your application.
+
+Lifecycle
+---------
+
+Every :class:`UIScreen` has a distinct lifecycle: *uninitialized*, *initialized*, *draw*,
+*process input* and *closed*. If the screen has already been initialized
+(it was *drawn* to a monitor) and will be shown again, then the screen skips
+the *uninitialized* stage. The screen can cycle between *draw* and *process input* stages by
+calling the :meth:`UIScreen.redraw` method after processing user input.
+
+In case :meth:`UIScreen.redraw` won't be called and no new screen is pushed, or this screen
+wasn't closed, then an application will stay in an infinite loop waiting for something to happen.
+This is correct behavior because there could be something in an event loop which will
+call :meth:`UIScreen.redraw` later.
+
+Rendering widgets
+-----------------
+
+The :meth:`UIScreen.refresh` method is the most important part of the :class:`UIScreen`.
+It contains preparations for rendering (creating widgets and adding them to containers).
+The :meth:`UIScreen.refresh` method will be called before anything is drawn on a monitor.
+
+The :attr:`UIScreen.window` attribute, which is the
+:class:`WindowContainer <simpleline.render.containers.WindowContainer>` instance, contains
+all items (widgets, containers) which are to be rendered by the screen.
+A new :class:`WindowContainer <simpleline.render.containers.WindowContainer>` is created in the
+:meth:`UIScreen.refresh` method for every screen redraw. The :attr:`UIScreen.title` attribute is
+passed to the :attr:`WindowContainer.title <simpleline.render.containers.WindowContainer.title>`
+property, and a developer can add :ref:`widgets <widgets_label>` and other
+:ref:`containers <containers_label>` to present items to a user by calling
+:meth:`WindowContainer.add() <simpleline.render.containers.WindowContainer.add>` or
+:meth:`WindowContainer.add_with_separator() <simpleline.render.containers.WindowContainer.add_with_separator>`.
+Multiple items can be added by calling these methods repeatedly.
+
+When everything is prepared properly in the :meth:`UIScreen.refresh` method, it needs to be drawn
+on the monitor for a user. This is handled by the :meth:`UIScreen.show_all` method. This method
+works automatically, but it could be useful for a developer, especially when the screen will not
+process input. In this case, the :meth:`UIScreen.input` is not called at all. Developers can
```

Comment:
`the UIScreen.input is not` -> either `the UIScreen.input method is not` or `UIScreen.input is not`